### PR TITLE
[Refactor:System] Remove Lock Date on Gradeables

### PR DESCRIPTION
**Pull request by cjreed121**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Right now a column exists called g_grade_locked_date. This doesn't have any functionality except to be able to be set. There is also no way an instructor could set this field.

### What is the new behavior?
This column has been removed from the database and site code.

### CODE REVIEW

**Comment by bmcutler** [comment]
Please look through your line changes, you've made edits to other fields.
Also there's a stray 'z' on one line.

**Comment by cjreed121** [comment]
> Please look through your line changes, you've made edits to other fields.
> Also there's a stray 'z' on one line.

Yes I missed the NOT NULL removal which I just fixed. I'm guessing the z you are referring to is on "timestamptz" which is intentional. "timestamptz" is timestamp with timezone in postgres. I can change it to "timestamp with timezone" if preferred.

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -5133,8 +5130,6 @@ public function createGradeable(\app\models\gradeable\Gradeable $gradeable) {
               g_ta_view_start_date,
               g_grade_start_date,
               g_grade_due_date,
-              g_grade_released_date,
```
Comment (MasterOdin) [inline site/app/libraries/database/DatabaseQueries.php:5136]: we probably want to keep this line?

File: migration/migrator/data/course_tables.sql [hunk 2]
```diff
@@ -281,12 +281,9 @@ CREATE TABLE public.gradeable (
     g_grade_start_date timestamp(6) with time zone NOT NULL,
     g_grade_due_date timestamp(6) with time zone NOT NULL,
     g_grade_released_date timestamp(6) with time zone NOT NULL,
-    g_grade_locked_date timestamp(6) with time zone,
     g_min_grading_group integer NOT NULL,
     g_syllabus_bucket character varying(255) NOT NULL,
     CONSTRAINT g_grade_due_date CHECK ((g_grade_due_date <= g_grade_released_date)),
-    CONSTRAINT g_grade_locked_date_max CHECK ((g_grade_locked_date <= '9999-03-01 00:00:00-05'::timestamp with time zone)),
-    CONSTRAINT g_grade_released_date CHECK ((g_grade_released_date <= g_grade_locked_date)),
```
Comment (MasterOdin) [inline migration/migrator/data/course_tables.sql:289]: These constraints are not removed as part of the migration?

**Review by MasterOdin — CHANGES_REQUESTED** [review]

File: site/app/libraries/database/DatabaseQueries.php (hunk 1 above)
Comment (cjreed121) [reply site/app/libraries/database/DatabaseQueries.php:5136]: Yes. Not sure why that got removed but will fix it.

File: migration/migrator/data/course_tables.sql (hunk 2 above)
Comment (cjreed121) [reply migration/migrator/data/course_tables.sql:289]: Those got removed after running the migration that I have added here. Or do I need to add something else to the migration?

**Review by bmcutler — APPROVED** [review]

**Pull request merged**